dts/bindings/rtc/st,stm32-rtc.yaml

```diff
@@ -18,3 +18,8 @@ properties:
     enum:
       - 1
       - 512
+
+  alarms-count:
+    type: int
+    description: |
+      Alarm count
```

Comment:
Is this alarm-count  a nb of alarm in the RTC instance like alarm A and/or alarm B ?
Only a very limited stm32 devices have one alarm and that can be identified by RTC Status Reg bits ALRAF  ALRBF

Reply:
This is the number of alarms. Alarms are indexed from 0 in the RTC API, if the device has 1 alarm, be it A or B, the count is 1, and the index used in the API is 0. Which alarm, A or B, maps to index 0 is not important from the application (as long as the index is consistent of course)

Reply:
It actually matters which alarm maps to index 0 and which to index 1. For example:

`if (id == 0) {
		LL_RTC_ALMA_Disable(RTC);`
`if (id == 1) {
		LL_RTC_ALMB_Disable(RTC);`

`alarms-count` was meant to denote how many alarms are present in the RTC instance and index 0 was meant to map to alarm A, index 1 to alarm B and in case of any STM32 devices coming out with more alarms than two, index 2 would map to alarm C and so on, and so on.

Wasn't sure how to handle this elegantly enough and I don't know if there are even any plans for devices with more than two alarms.